dot/peerset/peerset.go

```diff
@@ -294,15 +282,19 @@ func (ps *PeerSet) updateTime() error {
 					continue
 				}
 
-				lastDiscoveredTime := ps.peerState.lastConnectedAndDiscovered(set, peerID)
+				lastDiscoveredTime, err := ps.peerState.lastConnectedAndDiscovered(set, peerID)
+				if err != nil {
+					return fmt.Errorf("cannot get last connected peer: %w", err)
+				}
+
 				if lastDiscoveredTime.Add(forgetAfterTime).Second() >= currTime.Second() {
 					continue
 				}
 
 				// forget peer removes the peer from the list of members of the set.
 				err = ps.peerState.forgetPeer(set, peerID)
 				if err != nil {
-					return err
+					return fmt.Errorf("cannot forget peer %s from set %d: %w", peerID, set, err)
```

Comment:
supernit ideally the `forgetPeer` function should add the context about `set` and `peerID`. But this is fine too, just a tiny advice for future stuff.

Reply:
I didn't understand correctly what you mean by `the context about "set" and "peerID"`, is this related to the error message?

Reply:
Yeah about the error wrapping.

For example if you pass argument `arg1` to function `F`, `F` should wrap a sentinel error together with the context it has, in this case `arg1`. The caller of `F` should not wrap the error returned by `F` with context passed to it, but only with context of itself.

Reply:
Got it! update to just `fmt.Errorf("cannot forget peer: %w", err)`

Reply:
@EclesioMeloJunior I don't think this is done for that particular line :wink: